src/shogun/regression/GLM.h

```diff
@@ -0,0 +1,112 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Ahmed Khalifa
+ */
+/*
+References:
+https://www.rdocumentation.org/packages/stats/versions/3.6.2/topics/family
```

Comment:
I'd put this into the class docs.
But actually, don't just blindly put references, we dont really need reference to other libs in here. Rather a reference to wikipedia describing GLM (or Poisson regression) in general, in a conceptual/mathy sense